cmd/sync-tags/gomod.go

```diff
@@ -71,46 +75,46 @@ func updateGomodWithTaggedDependencies(tag string, depsRepo []string) (bool, err
 			return changed, fmt.Errorf("error running go mod download for %s: %v", depPkg, err)
 		}
 
-		// check if we have the pseudoVersion published already. if we don't, package it up
+		// check if we have the pseudoVersion/tag published already. if we don't, package it up
 		// and save to local mod download cache.
-		if err := packageDepToGoModCache(depPath, depPkg, rev, pseudoVersion, commitTime); err != nil {
+		if err := packageDepToGoModCache(depPath, depPkg, rev, pseudoVersionOrTag, commitTime); err != nil {
 			return changed, fmt.Errorf("failed to package %s dependency: %v", depPkg, err)
 		}
 
-		requireCommand := exec.Command("go", "mod", "edit", "-fmt", "-require", fmt.Sprintf("%s@%s", depPkg, pseudoVersion))
+		requireCommand := exec.Command("go", "mod", "edit", "-fmt", "-require", fmt.Sprintf("%s@%s", depPkg, pseudoVersionOrTag))
 		requireCommand.Env = append(os.Environ(), "GO111MODULE=on")
 		requireCommand.Stdout = os.Stdout
 		requireCommand.Stderr = os.Stderr
 		if err := requireCommand.Run(); err != nil {
-			return changed, fmt.Errorf("unable to pin %s in the require section of go.mod to %s: %v", depPkg, pseudoVersion, err)
+			return changed, fmt.Errorf("unable to pin %s in the require section of go.mod to %s: %v", depPkg, pseudoVersionOrTag, err)
 		}
 
-		replaceCommand := exec.Command("go", "mod", "edit", "-fmt", "-replace", fmt.Sprintf("%s=%s@%s", depPkg, depPkg, pseudoVersion))
+		replaceCommand := exec.Command("go", "mod", "edit", "-fmt", "-replace", fmt.Sprintf("%s=%s@%s", depPkg, depPkg, pseudoVersionOrTag))
 		replaceCommand.Env = append(os.Environ(), "GO111MODULE=on")
 		replaceCommand.Stdout = os.Stdout
 		replaceCommand.Stderr = os.Stderr
 		if err := replaceCommand.Run(); err != nil {
-			return changed, fmt.Errorf("unable to pin %s in the replace section of go.mod to %s: %v", depPkg, pseudoVersion, err)
+			return changed, fmt.Errorf("unable to pin %s in the replace section of go.mod to %s: %v", depPkg, pseudoVersionOrTag, err)
 		}
 
 		downloadCommand2 := exec.Command("go", "mod", "download")
 		downloadCommand2.Env = append(os.Environ(), "GO111MODULE=on", fmt.Sprintf("GOPRIVATE=%s", depPackages), "GOPROXY=https://proxy.golang.org")
 		downloadCommand2.Stdout = os.Stdout
 		downloadCommand2.Stderr = os.Stderr
 		if err := downloadCommand2.Run(); err != nil {
-			return changed, fmt.Errorf("error running go mod download for pseudo-version %s for %s: %v", pseudoVersion, depPkg, err)
+			return changed, fmt.Errorf("error running go mod download for pseudo-version %s for %s: %v", pseudoVersionOrTag, depPkg, err)
 		}
 
 		tidyCommand := exec.Command("go", "mod", "tidy")
-		tidyCommand.Env = append(os.Environ(), "GO111MODULE=on", "GOPOXY=file://${GOPATH}/pkg/mod/cache/download")
+		tidyCommand.Env = append(os.Environ(), "GO111MODULE=on")
```

Comment:
why is this not necessary anymore?

Reply:
GOPROXY is not needed while running go mod tidy. Moreover, the original code had a typo "GOPOXY", so it was never really used. :)

Reply:
> GOPROXY is not needed while running go mod tidy

really? tidy requires resolving dependencies, and this module has dependencies on unpublished tags/shas... I don't understand how this is working without GOPROXY pointing at our cache

Reply:
are we inheriting a GOPROXY setting from our parent env?

Reply:
> are we inheriting a GOPROXY setting from our parent env?

No, I don't think so. I tried running this locally as well and it did not globally try setting GOPROXY fwiw.

I _think_ `go mod tidy` looks at the local mod cache before donwloading from vcs...

Running the following commands at client-go checked out at `v0.18.0-alpha.0`:

- With the local mod cache cleared:

```
➜  client-go git:(2768422194b) GO111MODULE=on go mod tidy -v                                                                                                                                               0s 
get "k8s.io/apimachinery": found meta tag get.metaImport{Prefix:"k8s.io/apimachinery", VCS:"git", RepoRoot:"https://github.com/kubernetes/apimachinery"} at //k8s.io/apimachinery?go-get=1
go: k8s.io/apimachinery@v0.18.0-alpha.0: reading k8s.io/apimachinery/go.mod at revision v0.18.0-alpha.0: unknown revision v0.18.0-alpha.0
```

- After adding a `v0.18.0-alpha.0.mod` at `${GOPATH}/pkg/mod/cache/download/k8s.io/apimachinery/@v/`, `GO111MODULE=on go mod tidy -v` does not produce any output/errors.

Reply:
> With the local mod cache cleared:
> ...
> go: k8s.io/apimachinery@v0.18.0-alpha.0: reading k8s.io/apimachinery/go.mod at revision v0.18.0-alpha.0: unknown revision v0.18.0-alpha.0

did that fail the tidy command? did that fail our sync-tags command? (it should)

> After adding a v0.18.0-alpha.0.mod at ${GOPATH}/pkg/mod/cache/download/k8s.io/apimachinery/@v/, GO111MODULE=on go mod tidy -v does not produce any output/errors.

are we currently creating the `v0.18.0-alpha.0.mod` file?

Reply:
> did that fail the tidy command?

When I manually cleared my local cache and ran `go mod tidy`, yes.

> did that fail our sync-tags command? (it should)
> are we currently creating the v0.18.0-alpha.0.mod file?

We publish the `.mod`, `.info`, `.zip` files for each tag at the local cache, so we have a `v0.18.0-alpha.0.mod` file. And since we have this file, the sync-tags command does not fail.

https://github.com/kubernetes/publishing-bot/blob/3f3695e10c68b22812723076d843d2e48a65feaf/cmd/sync-tags/gomod.go#L80

Reply:
ok, so it sounds like we're getting the implicit behavior of "check the cache dir, then hit the network", and we happen to have the things we want in the cache dir.

I still think we should formalize that by pointing GOPROXY at the cache dir (fixing the typo in the envvar) rather than removing it. If the item is not in the cache dir, I really don't want us talking to the network to try to find things.

Reply:
> I still think we should formalize that by pointing GOPROXY at the cache dir (fixing the typo in the envvar, rather than removing it).

Done.